File: junit-examples/junit-jupiter-starter-maven/src/test/java/com/example/project/HumanCalculatorTests.java
package com.example.project;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

@Tag("human")
class HumanCalculatorTests {

	@Test
	void humanOnePlusOneEqualsTwo() {
		Calculator calculator = new Calculator();
		assertEquals(2, calculator.add(1, 1), "1 + 1 should equal 2");
	}

	@ParameterizedTest(name = "human {0} + {1} = {2}")
	@CsvSource({
			"0, 1, 1",
			"1, 2, 3",
			"49, 51, 100",
			"1, 100, 101"
	})
	void humanAdd(int first, int second, int expectedResult) {
		Calculator calculator = new Calculator();
		assertEquals(expectedResult, calculator.add(first, second),
				first + " + " + second + " should equal " + expectedResult);
	}

	@Test
	void humanAddHandlesNegative() {
		Calculator calculator = new Calculator();
		assertEquals(-1, calculator.add(2, -3), "2 + (-3) should equal -1");
	}
}

File: junit-examples/junit-jupiter-starter-sbt/src/test/scala/com/example/project/HumanCalculatorTests.scala
package com.example.project

import org.junit.jupiter.api.Assertions._
import org.junit.jupiter.api.Tag
import org.junit.jupiter.api.Test
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.CsvSource

@Tag("human")
class HumanCalculatorTests {

  @Test
  def humanOnePlusOneEqualsTwo(): Unit = {
    val calculator = new Calculator
    assertEquals(2, calculator.add(1, 1), "1 + 1 should equal 2")
  }

  @ParameterizedTest(name = "human {0} + {1} = {2}")
  @CsvSource(
    Array(
      "0, 1, 1",
      "1, 2, 3",
      "49, 51, 100",
      "1, 100, 101"
    )
  )
  def humanAdd(first: Int, second: Int, expected: Int): Unit = {
    val calculator = new Calculator
    assertEquals(expected, calculator.add(first, second), s"$first + $second should equal $expected")
  }

  @Test
  def humanDivisionByZeroError(): Unit = {
    val calculator = new Calculator
    val thrown = assertThrows(classOf[IllegalArgumentException], () => calculator.div(1, 0))
    assertTrue(thrown.getMessage.contains("Division by zero"))
  }
}
